test/expect.js

```diff
@@ -602,6 +602,18 @@ describe('expect', function () {
     expect({ foo: 1, bar: 2 }).to.not.have.all.keys(['baz', 'foo']);
     expect({ foo: 1, bar: 2 }).to.not.contain.all.keys(['baz', 'foo']);
 
+
+    var keys_maintain_original_order_issue359 = function () {
+      var expected = [ 'b', 'a' ];
+      var original_order = [ 'b', 'a' ];
+      var obj = { "b": 1, "a": 1 };
+      expect(expected).deep.equal(original_order);
+      expect(obj).keys(original_order);
+      expect(expected, "sorting happened unexpectedly in keys()")
+        .deep.equal(original_order);
+    }
+    keys_maintain_original_order_issue359();
+
```

Comment:
Sorry @gregglind, I don't think I made myself clear in the last message. There is no need for these assertions to be made inside a new function scope at all. Line 606, 614 and 615 can be completely removed, so as to keep the code inline with the other assertions - at the same indentation level. Alternatively if you feel like separating the test, it should go in a new `it()` e.g.

```js
it('keys(array) does not mutate array'. function () {
...
})
```

If you do a new `it()` test, it will need to be on the same indentation level as the other `it()`s.